Bottom: 626fa6167c67ba142887a5dfa93471065387bc60
Top:    60d08721a7b86ac23c646b71349dd1940bd75182
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-06-02 17:16:18 +0800

Refresh of refresh-of-tuntap-introduce-tx

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 6ab12b1..5cc6159 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -519,7 +519,6 @@ static struct tun_struct *tun_enable_queue(struct tun_file *tfile)
 static void tun_queue_purge(struct tun_file *tfile)
 {
 	struct sk_buff *skb;
-
 	while ((skb = skb_array_consume(&tfile->tx_array)) != NULL) {
 		kfree_skb(skb);
 	}
@@ -1721,7 +1720,7 @@ static int tun_peek(struct socket *sock, bool exact)
 		return 0;
 
 	if (tun->flags & IFF_TX_ARRAY) {
-		return skb_array_peek_len(&tfile->tx_array);
+		ret = skb_array_peek_len(&tfile->tx_array);
 	} else {
 		struct sk_buff *head;
